chore: Add pull request template for consistent contributions

MaxLee-dev opened the pull request:
<!-- 
Please follow the Conventional Commits specification for the PR title. (e.g., feat(component): Add Button component)

All sections in this template are optional.  
Feel free to remove sections that are not relevant to your pull request. 
-->
## 📝 Description of Changes

This pull request introduces a new pull request template to standardize the process of submitting changes and ensure quality and consistency across contributions.

### Changes to PR submission process:
- Added sections for related issues, a description of changes, screenshots for UI updates, and a checklist to ensure adherence to best practices like following Conventional Commits, adding tests, and updating documentation.
- Template location: [.github/PULL_REQUEST_TEMPLATE.md](diffhunk://#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35R1-R41)

## ✅ Checklist

Before submitting the PR, please check all of the following:

- [x] The PR title follows the Conventional Commits convention. (e.g., feat, fix, docs, style, refactor, test, chore)
- [ ] I have added tests for my changes.
- [ ] I have updated the Storybook or relevant documentation.
- [x] I have performed a self-code review.
- [x] I have read and followed the [Vapor Ground Rules](https://github.com/goorm-dev/vapor-ui/blob/main/.gemini/styleguide.md).

ZeroChoi2781 commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,45 @@
+
+All sections in this template are optional.
+Feel free to remove sections that are not relevant to your pull request.
+-->
+
+## 🔗Related Issues
+
+<!-- Please add any related issue numbers or links. -->
+
+- (e.g., Fixes #123)
+
+- (e.g., Notion: Design System Meeting Notes)
+
+## 📝 Description of Changes
+
+<!-- Please provide a brief summary of the changes in this PR. -->
+
+- (e.g., Added new Avatar component)
+
+- (e.g., Updated Color tokens to support dark mode)
+
+- (e.g., Fixed styles for the disabled state in the Input component)
+
+## 📸 Screenshots
+
+<!-- If there are any UI changes, please attach "before" and "after" screenshots. -->
+
+### Before
+
+### After
+
+## ✅ Checklist
+
+Before submitting the PR, please make sure you have checked all of the following items.
+
+[ ] The PR title follows the Conventional Commits convention. (e.g., feat, fix, docs, style, refactor, test, chore)
+
+[ ] I have added tests for my changes.
+
+[ ] I have updated the Storybook or relevant documentation.
```
@MaxLee-dev 

It would be nice to keep the English PR template to reflect our policy, and add `changeset` to the checklist !

MaxLee-dev commented:
> Comment out example lines: How about we add all (e.g., ...) lines, which are currently examples, as comments? There's a chance that these example lists might remain in sections where no content needs to be added, which could be confusing.
-  Changed all the (e.g., ...) examples are now in comments instead of bullet points
> Refine screenshot section: For screenshots, what do you think about guiding "Before" and "After" using only comments, and removing the headings? If we're adding a completely new feature, there wouldn't be a "Before" state, and we might end up submitting a PR with an empty heading.
- The template is now cleaner and more adaptable - contributors won't have empty headings for new features
> Address style guide length and language: The checklist currently asks to read the style guide, but I'm a bit concerned about its length and the fact that it's in Korean. It might be overwhelming for contributors.
- I agree. So I changed some wording to help contributors feel less overwhelmed. We should also translate our code guide into English as soon as possible.

ZeroChoi2781 commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,40 @@
+
+[ ] I have updated the Storybook or relevant documentation.
+
+[ ] I have added a changeset for this change (if applicable).
```
```suggestion
[ ] I have added a changeset for this change. (e.g., for any changes that affect users, such as component prop changes or new features)
```

Adding additional explanations to items will help contributors understand more clearly when to add changesets.
